### PR TITLE
Refactor | Add Network Configuration Provider

### PR DESCRIPTION
**Pull request by Da-Colon**
## Description
This PR only covers the setup and doesn't actually implement this code anywhere. 

The next PR will:
- Remove contract addresses from envs
- Remove `useAddresses` hook and all its references
- adds this provider to `App.tsx`
- updates `useSafeContracts` to use this config
- updates SDK url to use config's safeBaseURL in `useGnosisApiServices`

## Notes
One other thing a follow PR will take care of is updating the values for the contract addresses to use the `fractal-contracts` deployment contract address once that has made its way into the codebase

### CODE REVIEW

File: src/providers/NetworkConfig/NetworkConfigProvider.tsx [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    zodiacModuleProxyFactory: string;
+    linearVotingMasterCopy: string;
+    gnosisMultisend: string;
+    usulMasterCopy: string;
+    fractalModuleMasterCopy: string;
+    fractalNameRegistry: string;
+    votesTokenMasterCopy: string;
+    claimingFactory: string;
+    claimingMasterCopy: string;
+  };
```
Comment (Da-Colon) [inline src/providers/NetworkConfig/NetworkConfigProvider.tsx:19]: is there any other configuration value we want to add here?

File: src/providers/NetworkConfig/NetworkConfigProvider.tsx [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    default:
+      return defaultState;
+  }
+};
+
+export function NetworkConfigProvider({ children }: { children: ReactNode }) {
+  const {
+    state: { chainId },
+  } = useWeb3Provider();
+  const [config, setConfig] = useState<NetworkConfig>(getNetworkConfig(chainId));
```
Comment (Da-Colon) [inline src/providers/NetworkConfig/NetworkConfigProvider.tsx:61]: configuration should never be undefined

File: src/utils/percentFormat.spec.ts [hunk 3]
```diff
@@ -0,0 +1,98 @@
+import { BigNumber } from 'ethers';
```
Comment (herbig) [inline src/utils/percentFormat.spec.ts:1]: @Da-Colon why does this PR have a bunch of changes from @mudrila 's PR?

Comment (Da-Colon) [reply src/utils/percentFormat.spec.ts]: fixed. yeah rebase on develop forgetting that this one is based on refactor/sdk-swap-out

File: src/providers/NetworkConfig/networks/goerli.ts [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+export const goerliConfig: NetworkConfig = {
+  safeBaseURL: 'https://safe-transaction-goerli.safe.global',
+  contracts: {
+    gnosisSafe: '0xd9Db270c1B5E3Bd161E8c8503c55cEABeE709552',
+    gnosisSafeFactory: '0xa6B71E26C5e0845f74c812102Ca7114b6a896AB2',
+    zodiacModuleProxyFactory: '0x740020d3B1BF3E64e84dbA7175fC560B85EdB9bC',
+    linearVotingMasterCopy: '0x948db5691cc97AEcb4fF5FfcAEb72594B74D9D52',
+    gnosisMultisend: '0x40A2aCCbd92BCA938b02010E17A5b8929b49130D',
+    usulMasterCopy: '0xCdea1582a57Ca4A678070Fa645aaf3a40c2164C1',
+    fractalModuleMasterCopy: '0x260BcebDb25fa17f63972010E7b4EfC5C1D0fE0f',
```
Comment (herbig) [inline src/providers/NetworkConfig/networks/goerli.ts:12]: can these addresses come from the NPM package @christopherdancy set up?

**Review by herbig — APPROVED** [review]

File: src/providers/NetworkConfig/networks/goerli.ts (hunk 4 above)
Comment (Da-Colon) [reply src/providers/NetworkConfig/networks/goerli.ts:12]: I don't believe deployments are in the current release. when they are yeah we can easily update to reflect those addresses

**Pull request merged**